### ISSUE TITLE
Add RawSwap Augmentation to potential data augmentations

### ISSUE THREAD

**Issue by mayhemsloth** [mayhemsloth/Drum-Tabber#25]
Recently I was thinking about how the information encoded in images is roughly the same if you were to randomly swap any given pixels. If you did this a bunch of times (too many times), at some point the images would be degraded enough that you can no longer correctly interpret the information that was originally contained in the images. However there may be a sweet spot in there that you can swap some raw image values and thereby creating a new form of data augmentation, teaching the model about "raw information swap invariance". This is very similar to DropOut augmentation, because it is basically randomly setting random pixels to a different value (instead of 0 in DropOut). 

The idea is to implement this idea with raw audio data. The audio I am using is mono, which means that the raw data is simply an ordered list of values. If you were to randomly swap some of those values, you should have roughly the same information encoded in that raw data. I'll call this augmentation RawSwap, It should have 2 or maybe 3 configuration parameters: 1) of course the probability (chance) that it will trigger at all during the training loop 2) the minimum percentage of values swapped and 3) the maximum percentage of values swapped. I will also need to check these ranges and ensure that the drum event information is *not* degraded enough. 

This issue is resolved when the RawSwap augmentation is implemented into the training loop (and checked to ensure class label consistency).
